src/java/org/apache/cassandra/index/sai/plan/QueryView.java

```diff
@@ -238,8 +250,7 @@ else if (!processedMemtables.contains(memtable))
                         // The IndexViewManager got the update about this sstable, but there is no index for the sstable
                         // (e.g. index was dropped or got corrupt, etc.). In this case retrying won't fix it.
                         if (index == null)
-                            throw new MissingIndexException(indexContext, "Index " + indexContext.getIndexName() +
-                                                                          " not found for sstable: " + sstable.descriptor);
+                            throw new MissingIndexException(indexContext, sstable.descriptor);
```

Comment:
Nit: could pass `sstable` directly, and simplify slightly `MissingIndexException` to only handle `SSTableReader` (I don't mind much if you prefer as it is, just pointing it out).

Reply:
I didn't want to take a reference to SSTableReader, when the only thing I really need is some identification of it. 
I reworked the code now so I even don't need to store the Descriptor in the exception.